exec-service-wasmer/src/wasmer_instance.rs

```diff
@@ -193,4 +230,25 @@ impl Instance for WasmerInstance {
     fn get_breakpoint_value(&self) -> u64 {
         get_breakpoint_value(&self.wasmer_instance)
     }
+
+    fn reset(&self) -> Result<(), String> {
+        self.wasmer_instance.reset()
+    }
+
+    unsafe fn cache(
+        &self,
+        cache_bytes_ptr: *mut *const u8,
```

Comment:
I think we should replace this with `cache_bytes: &mut [u8]`
I don't think we need to work with raw pointers in this interface.

Or, even better, we could return the serialized Vec<u8> and only write to memory like this in the c-api.
Does not even need to be unsafe in that case.

Reply:
good catch sir!

Reply:
done